_posts/2024-07-08-idea-editor.adoc

```diff
@@ -0,0 +1,138 @@
++
+image::/assets/img/idea-editor/oc-login.png[Copy login command]
+. Copy the displayed *oc login* command and execute it in your local terminal.
+
+How to use it?
+
+1. Use the User Dashboard to spin up a new or start an existing Workspace. When running a new Workspace, make sure the right editor is chosen:
++
+image::/assets/img/idea-editor/editor-selector.png[Editor Selector]
+{nbsp} +
+
+2. Wait for the prompt to open your local JetBrains Gateway application to appear:
++
+image::/assets/img/idea-editor/open-prompt.png[Open Gateway prompt]
+{nbsp} +
+
+3. Click the *Open Gateway* button to start your local IDEA connected to your CDE:
++
+image::/assets/img/idea-editor/connecting.png[Connecting to remote host]
+
+You can try it out at https://developers.redhat.com/developer-sandbox/ide[Red Hat Developer Sandbox]. Clicking on https://workspaces.openshift.com/f?url=https://github.com/che-samples/web-nodejs-sample&che-editor=https://eclipse-che.github.io/che-plugin-registry/main/v3/plugins/che-incubator/che-idea-server/next/devfile.yaml&policies.create=peruser[this URL] automatically starts a CDE on the Developer Sandbox with a sample project imported. Once the CDE is ready, you’ll be asked to start a local IDEA for connecting to the CDE.
+
+==== Option 2: from the JetBrains Gateway application
+
+This option is more suitable when you want to continue working with an existing CDE, without accessing the Eclipse Che User Dashboard. 
+
+IMPORTANT: When using the JetBrains Gateway application the creation of a new CDE is currently not allowed, and you can only connect to already-created development environments.
+
+This option has no additional prerequisites like having the local OpenShift CLI tool (*oc*) logged in your OpenShift cluster. But if you do log in, the Gateway application will use the CLI's login session which means you will not need to enter your login info from the Gateway application every time you use it.
+
+How to use it?
+
+1. Open the Gateway application and click the *Connect to Dev Spaces* button:
++
+image::/assets/img/idea-editor/gateway-window.png[Gateway window]
+
+2. In the next step, the Gateway plugin will try to detect if your local OpenShift CLI tool (*oc*) is logged into your OpenShift cluster to read the OpenShift API server connection parameters from the CLI config file. You can also provide the connection parameters manually:
++
+image::/assets/img/idea-editor/login-window.png[Login to API server]
++
+After clicking the *Check connection and continue* button, the plugin will try to read your existing CDEs.
+
+3. Choose an existing CDE and click the *Connect* button:
++
+image::/assets/img/idea-editor/select-ds.png[DevWorkspace selection]
++
+WARNING: Currently, all CDEs are listed regardless of the configured editor. As Eclipse Che doesn’t allow the user to switch the editor for an already created CDE, make sure the right CDE is chosen.
```

Comment:
@azatsarynnyy AFAIK it is possible to change IDE for existing CDE via URL parameter. It is not possible to do in the runtime though, so maybe `created` should be changed to `running` ?